Library/Tracking/KSRAnalytics.swift

```diff
@@ -1,72 +1,26 @@
   private let screen: UIScreenType
-
-  private enum DataLakeApprovedEvent: String, CaseIterable {
-    case activityFeedViewed = "Activity Feed Viewed"
-    case addNewCardButtonClicked = "Add New Card Button Clicked"
-    case addOnsContinueButtonClicked = "Add-Ons Continue Button Clicked"
-    case addOnsPageViewed = "Add-Ons Page Viewed"
-    case campaignDetailsButtonClicked = "Campaign Details Button Clicked"
-    case campaignDetailsPledgeButtonClicked = "Campaign Details Pledge Button Clicked"
-    case checkoutPaymentPageViewed = "Checkout Payment Page Viewed"
-    case collectionViewed = "Collection Viewed"
-    case continueWithAppleButtonClicked = "Continue With Apple Button Clicked"
-    case editorialCardClicked = "Editorial Card Clicked"
-    case explorePageViewed = "Explore Page Viewed"
-    case exploreSortClicked = "Explore Sort Clicked"
-    case fbLoginOrSignupButtonClicked = "Facebook Log In or Signup Button Clicked"
-    case filterClicked = "Filter Clicked"
-    case fixPledgeButtonClicked = "Fix Pledge Button Clicked"
-    case forgotPasswordViewed = "Forgot Password Viewed"
-    case loginButtonClicked = "Log In Button Clicked"
-    case loginOrSignupButtonClicked = "Log In or Signup Button Clicked"
-    case loginOrSignupPageViewed = "Log In or Signup Page Viewed"
-    case loginSubmitButtonClicked = "Log In Submit Button Clicked"
-    case managePledgeButtonClicked = "Manage Pledge Button Clicked"
-    case onboardingCarouselSwiped = "Onboarding Carousel Swiped"
-    case onboardingContinueButtonClicked = "Onboarding Continue Button Clicked"
-    case onboardingGetStartedButtonClicked = "Onboarding Get Started Button Clicked"
-    case onboardingSkipButtonClicked = "Onboarding Skip Button Clicked"
-    case pledgeSubmitButtonClicked = "Pledge Submit Button Clicked"
-    case projectCardClicked = "Project Card Clicked"
-    case projectPagePledgeButtonClicked = "Project Page Pledge Button Clicked"
-    case projectPageViewed = "Project Page Viewed"
-    case projectSwiped = "Project Swiped"
-    case searchPageViewed = "Search Page Viewed"
-    case searchResultsLoaded = "Search Results Loaded"
-    case selectRewardButtonClicked = "Select Reward Button Clicked"
-    case signupButtonClicked = "Signup Button Clicked"
-    case signupSubmitButtonClicked = "Signup Submit Button Clicked"
-    case skipVerificationButtonClicked = "Skip Verification Button Clicked"
-    case tabBarClicked = "Tab Bar Clicked"
-    case thanksPageViewed = "Thanks Page Viewed"
-    case twoFactorConfirmationViewed = "Two-Factor Confirmation Viewed"
-    case verificationScreenViewed = "Verification Screen Viewed"
-    case watchProjectButtonClicked = "Watch Project Button Clicked"
-
-    static func allApprovedEvents() -> [String] {
-      return DataLakeApprovedEvent.allCases.map { $0.rawValue }
-    }
-  }
+  private let segmentClient: TrackingClientType & IdentifyingTrackingClient & ScreenTrackingClient
```

Comment:
A couple of notes:

- I deleted the `ScreenTrackingClientType` and all of its references after discovering that Braze doesn't support `screen` method calls in its side-by-side integration with Segment (updated the engineering plan with this info).

-  We did this for testing reasons actually. The `TrackingClientType` and `IdentifyingTrackingClient` protocols
 allow us to create mocks to test these code paths. The protocol exposes functions that are named similarly
 but differently (`track` specifically) so that we can perform the check to see that tracking is enabled before calling any of the functions on the library itself so as to not unintentionally contribute to tracking data during debugging. Also, once we remove `DataLakeClient` we can very likely remove the `TrackingClientType` protocol since Segment will be handling all of the logic by itself. Let me know what you think?

Reply:
Sounds like a plan, thanks for the explaination